packages/binding-coap/src/coap-client.ts

```diff
@@ -53,83 +53,83 @@ export default class CoapClient implements ProtocolClient {
         let contentType = res.headers["Content-Format"];
         if (!contentType) contentType = form.contentType;
-        
+
         resolve({ type: contentType, body: res.payload });
-      });
-      req.on("error", (err: Error) => reject(err));
-      req.end();
+    });
+    req.on("error", (err: Error) => reject(err));
+    req.end();
     });
   }
 
   public writeResource(form: CoapForm, content: Content): Promise<any> {
     return new Promise<void>((resolve, reject) => {
 
-      let req = this.generateRequest(form, "PUT");
+    let req = this.generateRequest(form, "PUT");
 
-      // TODO set Content-FOrmat
+    // TODO set Content-FOrmat
 
-      console.debug("[binding-coap]",`CoapClient sending ${req.statusCode} to ${form.href}`);
+    console.debug("[binding-coap]",`CoapClient sending ${req.statusCode} to ${form.href}`);
 
-      req.on("response", (res: any) => {
+    req.on("response", (res: any) => {
         console.debug("[binding-coap]",`CoapClient received ${res.code} from ${form.href}`);
         console.debug("[binding-coap]",`CoapClient received headers: ${JSON.stringify(res.headers)}`);
         resolve();
-      });
-      req.on("error", (err: Error) => reject(err));
-      req.setOption("Content-Format", content.type);
-      req.write(content.body);
-      req.end();
+    });
+    req.on("error", (err: Error) => reject(err));
+    req.setOption("Content-Format", content.type);
+    req.write(content.body);
+    req.end();
     });
   }
 
   public invokeResource(form: CoapForm, content?: Content): Promise<Content> {
     return new Promise<Content>((resolve, reject) => {
 
-      let req = this.generateRequest(form, "POST");
+    let req = this.generateRequest(form, "POST");
 
-      console.debug("[binding-coap]",`CoapClient sending ${req.statusCode} to ${form.href}`);
+    console.debug("[binding-coap]",`CoapClient sending ${req.statusCode} to ${form.href}`);
 
-      req.on("response", (res: any) => {
+    req.on("response", (res: any) => {
         console.debug("[binding-coap]",`CoapClient received ${res.code} from ${form.href}`);
         console.debug("[binding-coap]",`CoapClient received Content-Format: ${res.headers["Content-Format"]}`);
         console.debug("[binding-coap]",`CoapClient received headers: ${JSON.stringify(res.headers)}`);
         let contentType = res.headers["Content-Format"];
         resolve({ type: contentType, body: res.payload });
-      });
-      req.on("error", (err: Error) => reject(err));
-      if (content) {
+    });
+    req.on("error", (err: Error) => reject(err));
+    if (content) {
         req.setOption("Content-Format", content.type);
         req.write(content.body);
-      }
-      req.end();
+    }
+    req.end();
     });
   }
 
   public unlinkResource(form: CoapForm): Promise<any> {
     return new Promise<void>((resolve, reject) => {
 
-      let req = this.generateRequest(form, "GET", false);
+    let req = this.generateRequest(form, "GET", false);
```

Comment:
not sure why but eclint does strange nesting.
Code in promise should be nested...